### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by benogle**
I've started digging into the autocomplete code to get an idea of what is happening in here. I plan on fixing some issues with it here and there and writing a couple providers to get a feel for the code and approach. 

So here are some surface changes that jumped out at me. The biggest changes are the removal of `theorist` and `event-kit`, which are not needed, or included in core.

### CODE REVIEW

**Comment by joefitzgerald** [comment]
These LGTM so far, thanks for the thorough review. @park9140 and I would like to add a lint config such that the code can follow the style guidelines the core team advocates. Do you have one?

**Comment by benogle** [comment]
> Do you have one?

No, we have @kevinsawicki

**Comment by joefitzgerald** [comment]
Haha :laughing:. Would you be open to helping @park9140 and I define one?

File: lib/fuzzy-provider.coffee [hunk 1]
```diff
@@ -1,8 +1,7 @@
 _ = require 'underscore-plus'
 Suggestion = require './suggestion'
 fuzzaldrin = require 'fuzzaldrin'
-{CompositeDisposable} = require 'event-kit'
-{TextEditor}  = require 'atom'
+{TextEditor, CompositeDisposable}  = require 'atom'
```
Comment (park9140) [inline lib/fuzzy-provider.coffee]: do we really want to rely on the composite disposable exposed from atom?  I'm not sure of a good argument for doing this.

File: package.json [hunk 2]
```diff
@@ -9,14 +9,12 @@
   },
   "dependencies": {
     "async": ">=0.9.0",
-    "event-kit": ">=0.8.1",
```
Comment (park9140) [inline package.json]: Can we just make this a peer dependency so we get the one installed by atom?

File: package.json [hunk 3]
```diff
@@ -9,14 +9,12 @@
...
     "fuzzaldrin": ">=1.0.0",
     "grim": ">=0.12.0",
     "minimatch": ">=0.2.14",
     "node-uuid": ">=1.4.1",
     "scoped-property-store": ">=0.16.2",
     "temp": ">=0.7.0",
```
Comment (park9140) [inline package.json]: do we need tem?

**Comment by park9140** [comment]
@benogle, looks good, just wondering if there is a specific reason to use the event-kit provided by core rather than having the dependency?

**Comment by lierdakil** [comment]
@park9140 for one, having a dependency forces installation of event-kit into package-specific node_modules, while requiring it from atom does not. Code de-duplication and all that. Not a big deal, but not something to shrug off either, IMO.

File: package.json (hunk 3 above)
Comment (joefitzgerald) [inline package.json]: temp is really a devDependency...

File: lib/fuzzy-provider.coffee (hunk 1 above)
Comment (joefitzgerald) [inline lib/fuzzy-provider.coffee]: They export it, so no reason not to instantiate based on that prototype. Of course, I've depended on plenty of things from atom that have then been deprecated, so YMMV :wink:.

File: package.json (hunk 2 above)
Comment (joefitzgerald) [inline package.json]: The less dependencies in package.json, the better.

**Comment by joefitzgerald** [comment]
@benogle Mind if I merge? Or do you want to do more review.

**Comment by benogle** [comment]
Go for it

**Comment by joefitzgerald** [comment]
@benogle Left the branch open in case you want to keep making changes.

**Comment by benogle** [comment]
K, i'll delete it

**Pull request merged**